### ISSUE TITLE
[swagger-codegen-3.0.0-rc0] Build and swagger-codegen-cli.jar execution issues

### ISSUE THREAD

**Issue by wol190** [swagger-api/swagger-codegen#7509]
##### Description
When I build swagger-codegen-3.0.0-rc0 using "mvn clean package" there are errors and also when I run /swagger-codegen-cli.jar generate

##### Swagger-codegen version
swagger-codegen-3.0.0-rc0 from https://github.com/swagger-api/swagger-codegen/releases/tag/v3.0.0-rc0

##### Swagger declaration file content or url: Cutdown Pet.yaml:
```yaml
openapi: "3.0.0"
info:
  version: 1.0.0
  title: Swagger Petstore
servers:
  - url: http://127.0.0.1:8080
paths:
  /pets:
    get:
      summary: Return the pet
      operationId: getThePet
      responses:
        '200':
          description: A pet
          content:
            application/json:    
              schema:
                $ref: "#/components/schemas/Pet"
components:
  schemas:
    Pet:
      required:
        - id
        - name
      properties:
        id:
          type: integer
          format: int64
        name:
          type: string
      example:
        id: 1
        name: "Trevor"
  ```

##### Command line used for generation
java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -l java -i ../pet.yaml -o C:/tmp/server/petstore3/java

##### Build errors (duplicate output lines removed)
mvn clean package
[INFO] Scanning for projects...
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Build Order:
[INFO]
[INFO] swagger-codegen-project
[INFO] swagger-codegen (core library)
[INFO] swagger-codegen (executable)
[INFO] swagger-codegen (maven-plugin)
[INFO] swagger-generator
[INFO] Building swagger-codegen-project 3.0.0-rc0
[INFO] --- maven-clean-plugin:2.5:clean (default-clean) @ swagger-codegen-project ---
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-maven) @ swagger-codegen-project ---
[INFO] --- maven-enforcer-plugin:1.4.1:enforce (enforce-versions) @ swagger-codegen-project ---
[INFO] --- maven-surefire-plugin:2.19.1:test (test-testng) @ swagger-codegen-project ---
[INFO] No tests to run.
[INFO] --- maven-dependency-plugin:2.8:copy-dependencies (default) @ swagger-codegen-project ---

...

[INFO] --- maven-surefire-plugin:2.19.1:test (default-test) @ swagger-codegen ---

-------------------------------------------------------
 T E S T S
Results :
Tests run: 0, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] --- maven-surefire-plugin:2.19.1:test (test-testng) @ swagger-codegen ---
Running TestSuite
[main] INFO org.reflections.Reflections - Reflections took 266 ms to scan 2 urls, producing 43 keys and 232 values
[main] INFO org.reflections.Reflections - Reflections took 159 ms to scan 2 urls, producing 43 keys and 232 values
[main] INFO io.swagger.codegen.ignore.CodegenIgnoreProcessor - No .swagger-codegen-ignore file found.
[main] WARN io.swagger.codegen.DefaultCodegen - generated unique operationId `duplicate_0`
[main] WARN io.swagger.codegen.DefaultCodegen - Empty operationId found for path: get /pet. Renamed to auto-generated operationId: petGet
[main] WARN io.swagger.codegen.ignore.CodegenIgnoreProcessor - Output directory does not exist, or is inaccessible. No file (.swagger-codegen-ignore) will be evaluated.
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property Pet
[main] ERROR io.swagger.codegen.DefaultCodegen - String to be sanitized is null. Default to Object
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property Order
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property User
the size is: 2
[main] WARN io.swagger.codegen.DefaultCodegen - escapeUnsafeCharacters should be overridden in the code generator with proper logic to escape unsafe characters
[main] WARN io.swagger.codegen.DefaultCodegen - escapeQuotationMark should be overridden in the code generator with proper logic to escape single/double quote
Tests run: 45, Failures: 0, Errors: 0, Skipped: 2, Time elapsed: 3.382 sec - in TestSuite

##### Execution errors running swagger-codegen-cli.jar generate
java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -l java -i ../pet.yaml -o C:/tmp/server/petstore3/java
[Thread-0] INFO io.swagger.codegen.ignore.CodegenIgnoreProcessor - No .swagger-codegen-ignore file found.
[Thread-0] ERROR io.swagger.codegen.DefaultCodegen - String to be sanitized is null. Default to Object
Exception in thread "Thread-0" java.lang.RuntimeException: Could not generate model 'Pet'
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:407)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:722)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:312)
        at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: /v2/Java/\model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
        at io.swagger.codegen.DefaultGenerator.getHandlebars(DefaultGenerator.java:1013)
        at io.swagger.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:738)
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:394)
        ... 3 more

**Comment by HugoMario** [comment]
@wol190 
when you have a chance, can you please check if this issue is still present?

**Comment by wol190** [comment]
@HugoMario Many thanks for the update and support on this issue. I took the latest 3.0.0 snapshot. There are still errors, but I'm happy to say that I can use  swagger-codegen-cli.jar and build the resulting output. However, I haven't as yet tried to use the resulting java client. 

Errors with mvn clean package
```
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property Pet
--
[main] ERROR io.swagger.codegen.DefaultCodegen - String to be sanitized is null. Default to Object
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property Order
[main] ERROR io.swagger.codegen.languages.AbstractJavaCodegen - No Type defined for Property User

```
Errors with: java -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate -l java -i ../pet.yaml -o C:/tmp/server/petstore3/java
(noting that the output is generated and can be built)
```
[Thread-0] ERROR io.swagger.codegen.DefaultCodegen - String to be sanitized is null. Default to Object
--
[Thread-0] ERROR io.swagger.codegen.DefaultGenerator - can't open v2/Java\gradle-wrapper.jar for input
```

**Comment by HugoMario** [comment]
@wol190 , most of them are actually `WARN`message, i'm going to update the log level. And fix that path with `gradle` file. 

Thanks.

**Comment by scarter93** [comment]
This is also happening to me for spring generation

`Caused by: java.io.FileNotFoundException: /JavaSpring/model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
        at io.swagger.codegen.DefaultGenerator.getHandlebars(DefaultGenerator.java:1014)
        at io.swagger.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:738)
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:394)`

**Comment by HugoMario** [comment]
@scarter93 , that happen because codegen `3.0.0` is supporting temporary only `java`and `inflector` languages.

For more details you can check [3.0.0-rc0 release](https://github.com/swagger-api/swagger-codegen/releases/tag/v3.0.0-rc0)

**Comment by HugoMario** [comment]
this issue is solved so closing now.

**Comment by soberich** [comment]
I have exactely this issue. 
From recent experience my immidiate guess is that it's because of Windows 7
I'll try on ubuntu and let know.

**Comment by anand188** [comment]
Hi ,
same error using pom 
<groupId>io.swagger</groupId>
  <artifactId>swagger-codegen-maven-plugin</artifactId>
  <version>3.0.0-rc0</version>
Caused by: java.io.FileNotFoundException: /v2/JavaJaxRS/\model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
        at io.swagger.codegen.DefaultGenerator.getHandlebars(DefaultGenerator.java:1013)
        at io.swagger.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:738)
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:394)
        ... 24 more

**Comment by Kiolali** [comment]
I have the same error on Windows 10

**Comment by kowlown** [comment]
I wonder how are the tests are made ? Same errors than the others. The issue is "solved" for which release. Do we have an ETA for the fix ?

**Comment by HugoMario** [comment]
hey @kowlown @aymhce @Kiolali , sorry for this late response.
A PR was created in order to solve this issue, i'll be checking it and let you know if the issue is solved with that.

**Comment by HugoMario** [comment]
The path issue has been solved for `3.0.0-SNAPSHOT`, so going to close this, but please, let us know if you still get the error.

**Comment by kowlown** [comment]
I still have an issue with the version **swagger-codegen-cli-3.0.0-20180608.091024-78**, I had a java.io.FileNotFoundException: /v2\Java/model.mustache. 

To solve the issue I have to extract the v2 directory from the jar and reference the template with --template-dir parameter. So there is still an issue with classpath directory.

I am on Windows 10, with Java 8 (1.8.0_121)
Executed the jar with both CMD and the bash of MINGW.

**Comment by soerenGer** [comment]
I have just checked out the current 3.0.0 snapshot version, running on Windows 10 (64bit) and java 1.8.0 and can confirm that the current cutting-edge branch still has the problem with Windows paths.

```
Exception in thread "Thread-1" java.lang.RuntimeException: Could not generate model 'CompanyCountry'
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:409)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:729)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:331)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: /v2\Java/model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
[...]

```

Building and testing with maven worked without any glitch, creating code however stops here. I tried the trick from @kowlown which worked for me as well, but of course that's a ridiculous hack.

**Comment by HugoMario** [comment]
hey @kowlown sorry i gave you the wrong one.

@soerenGer , can you please try with this version [swagger-codegen-cli-3.0.0-20180615.112032-79.jar](https://oss.sonatype.org/content/repositories/snapshots/io/swagger/swagger-codegen-cli/3.0.0-SNAPSHOT/swagger-codegen-cli-3.0.0-20180615.112032-79.jar), and let me know how it goes?

**Comment by soerenGer** [comment]
Hi @HugoMario, just tried, still the same problem. I really made it super-simple: Created fresh directory only with my yaml, d'ed your jar, started the generator, done.
```

Exception in thread "Thread-1" java.lang.RuntimeException: Could not generate model 'CompanyCountry'
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:409)
        at io.swagger.codegen.DefaultGenerator.generate(DefaultGenerator.java:729)
        at io.swagger.codegen.cmd.Generate.run(Generate.java:331)
        at java.lang.Thread.run(Unknown Source)
Caused by: java.io.FileNotFoundException: /v2\Java/model.mustache
        at com.github.jknack.handlebars.io.URLTemplateLoader.sourceAt(URLTemplateLoader.java:70)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:357)
        at com.github.jknack.handlebars.Handlebars.compile(Handlebars.java:343)
        at io.swagger.codegen.DefaultGenerator.getHandlebars(DefaultGenerator.java:1027)
        at io.swagger.codegen.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:745)
        at io.swagger.codegen.DefaultGenerator.generateModels(DefaultGenerator.java:396)
        ... 3 more
```

**Comment by HugoMario** [comment]
@soerenGer what OS are you using? independently of that, i think we need to reopen this issue. But please let me know, and if you can share you spec or part of it, it would be helpful too.

**Comment by soerenGer** [comment]
Hi, I am using Windows 10, 64 bit. What "spec" do you need? I could provide you with a minimal yaml file that can reproduce the problem.

**Comment by HugoMario** [comment]
Hey @soerenGer, whenn you have a chance, can you please check if the issue has been fixed for this version: [swagger-codegen-cli-3.0.0-20180616.031247-80.jar](https://oss.sonatype.org/content/repositories/snapshots/io/swagger/swagger-codegen-cli/3.0.0-SNAPSHOT/swagger-codegen-cli-3.0.0-20180616.031247-80.jar)??

**Comment by soerenGer** [comment]
Hi Mario, two thumbs up – this version works well, source code is generated as expected and also seems to work well. I hope this fix will make it now fast into the master ,-)

Thanks for fixing!
Soeren

**Comment by kowlown** [comment]
Thank you very much, the new jar fix the issue.

**Comment by yeikel** [comment]
Is there any release with this fix? @HugoMario

**Comment by srividhyak27** [comment]
I am using 3.0.0-rc1, have the same issue.  Is the fix/jar mentioned above available in a release 
Thanks,
Sri

**Comment by pcoltau** [comment]
I getting the same issue on Mac, but for swift4. 

I've tried 3.0.0-rc1 and also the latest snapshot: [swagger-codegen-cli-3.0.0-20180717.153005-90.jar](https://oss.sonatype.org/content/repositories/snapshots/io/swagger/swagger-codegen-cli/3.0.0-SNAPSHOT/swagger-codegen-cli-3.0.0-20180717.153005-90.jar)

**Comment by marcelobp** [comment]
Is <language>inflector</language> still available for codegen 3.0? Generation is not working for me if I use this option. It works with <language>java</language> though. Tks

**Comment by anandjaisy** [comment]
Getting the same issue on downloaded jar swagger-codegen-cli-3.0.34.jar with Angular application and Micronaut

```
java -jar tools/swagger-codegen-cli-3.0.34.jar generate \
  -i http://localhost:8080/swagger/fete-bird-api-gateway-0.0.yml \
  -l typescript-angular \
  -o /libs/service/src/lib/swagger-specification \
  -Dmodels -DmodelDocs=false \
  -c tools/angular-swagger-option.json
```

**Exception**

11:23:33.131 [Thread-0] WARN  i.s.c.v.i.CodegenIgnoreProcessor - Output directory does not exist, or is inaccessible. No file (.swagger-codegen-ignore) will be evaluated.
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.189 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.190 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.190 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.204 [Thread-0] WARN  i.s.c.v.g.t.AbstractTypeScriptClientCodegen - Object (model name matches existing language type) cannot be used as a model name. Renamed to ModelObject
11:23:33.257 [Thread-0] INFO  i.s.codegen.v3.AbstractGenerator - writing file /libs/service/src/lib/swagger-specification/model/activeDates.ts
Exception in thread "Thread-0" java.lang.RuntimeException: Could not generate model 'ActiveDates'
    at io.swagger.codegen.v3.DefaultGenerator.generateModels(DefaultGenerator.java:451)
    at io.swagger.codegen.v3.DefaultGenerator.generate(DefaultGenerator.java:782)
    at io.swagger.codegen.v3.cli.cmd.Generate.run(Generate.java:388)
    at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.io.FileNotFoundException: /libs/service/src/lib/swagger-specification/model/activeDates.ts (No such file or directory)
    at java.base/java.io.FileOutputStream.open0(Native Method)
    at java.base/java.io.FileOutputStream.open(FileOutputStream.java:293)
    at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:235)
    at java.base/java.io.FileOutputStream.<init>(FileOutputStream.java:184)
    at io.swagger.codegen.v3.AbstractGenerator.writeToFile(AbstractGenerator.java:32)
    at io.swagger.codegen.v3.DefaultGenerator.processTemplateToFile(DefaultGenerator.java:799)
    at io.swagger.codegen.v3.DefaultGenerator.generateModels(DefaultGenerator.java:438)
